### ISSUE TITLE
SOQL Mass Action with Flow Action / Batch Size=1 Executes Twice

### ISSUE THREAD

**Issue by JeroenSfdc** [douglascayers-org/sfdx-mass-action-scheduler#107]
## Summary
I'm having a Mass Configuration with a SOQL Source, and Flow as target Action. 
Batch Size = 1 as the Flow is sending out emails (through Apex invocable action) I want to isolate every source record.

`SELECT Id FROM SummaryReport__c WHERE Status__c = 'Scheduled'`

The Configuration is Scheduled on the top of every hour.

It works fine, and the Flows are being executed. But, for every Flow executed it also tries to invoke the Flow a 2nd time. The 2nd time the mapped Id is not passed into the Flow, and it errors out at the fist step. This is very consistent but it seems to happen only on our Production org. 

I verified:
- No more than a single Scheduled Apex, only one Scheduled Apex job for the Config Id exists.
- I figured, maybe it has to do with a deleted configuration, there are no deleted records in the recycle bin for either the `dca_mass_action__Mass_Action_Configuration__c` or `dca_mass_action__Mass_Action_Mapping__c` 

Since debugging a managed package is limited, I was yet unable to identify the root-cause.

```
`Flow Details
Flow API Name: Distribute_Summary_Reports_Batch
Type: Autolaunched Flow
Version: 1
Status: Active
Org: IMCD Group (xxxx)
Flow Interview Details
Interview Label: Distribute Summary Reports Batch 7/8/2019 11:00 PM
Current User: Integration User (xxxx)
Start time: 7/8/2019 11:00 PM
Duration: 0 seconds
How the Interview Started
Integration User (xxxx) started the flow interview.
FAST LOOKUP: getSummaryReport
Find all SummaryReport__c records where:
Id Equals {!recordId} (null)
Store those records in {!summaryReport}.
Save these field values in the variable: Id, Status__c, ScheduledDt__c, Sent__c, SentDt__c, Language__c, TF_AccountIMCDCompany__c, CreatedById
Result
Failed to find records.`
```

## Version of Mass Action Scheduler
2.2

## Steps To Reproduce
Not possible yet. I can grant access to our org if needed. 

## Expected Behavior
Flow should be executed once, and only once. 

## Actual Result
Exception email received for the flow. 

## Source Type
- [ ] Reports
- [ ] List Views
- [X] SOQL Queries

## Action Type
- [X] Process Builder & Flows
- [ ] Workflow Rules
- [ ] Email Alerts
- [ ] Quick Actions
- [ ] Apex

## Does your configuration use Named Credentials?
- [X] Yes
- [ ] No

But Scheduled Job is owned by another user.

**Comment by douglascayers** [comment]
Hi @JeroenSfdc, that sounds very strange indeed. And you say this only happens in your production org, you can't reproduce it in a sandbox?

Do you get the same results when not using a Named Credential?

**Comment by JeroenSfdc** [comment]
Hey @douglascayers. I've changed the MAS Job to not use the named credential, but temporarily run under my account. I can confirm that the apex Batch Job executes under my credentials. I introduced also a safety step in the Flow to log an error to our ```DebugLog__c``` object in case the Flow receives an empty ```recordId```. Now the interesting bit here is, that the ```DebugLog__c``` is created by the user tied to our MAS named credential. So something definitely is not right, but I'm unable to figure out yet what.
